pyjetty/alice_analysis/analysis/user/substructure/run_analysis.py

```diff
@@ -495,94 +579,131 @@ def plot_systematic_uncertainties(self, jetR, obs_label, obs_setting, grooming_s
     text_latex.SetNDC()
     text = str(min_pt_truth) + ' < #it{p}_{T, ch jet} < ' + str(max_pt_truth)
     text_latex.DrawLatex(0.3, 0.85, text)
-    
+
     text_latex = ROOT.TLatex()
     text_latex.SetNDC()
     text = '#it{R} = ' + str(jetR)
     text_latex.DrawLatex(0.3, 0.78, text)
-    
+
     subobs_label = self.utils.formatted_subobs_label(self.observable)
     if subobs_label:
       text = '{} = {}'.format(subobs_label, obs_setting)
       text_latex.DrawLatex(0.3, 0.71, text)
-    
+
     if grooming_setting:
       text = self.utils.formatted_grooming_label(grooming_setting)
       text_latex.DrawLatex(0.3, 0.64, text)
-      
+
     output_dir = getattr(self, 'output_dir_systematics')
-    outputFilename = os.path.join(output_dir, 'hSystematics_R{}_{}_{}-{}{}'.format(self.utils.remove_periods(jetR), obs_label, int(min_pt_truth), int(max_pt_truth), self.file_format))
+    outputFilename = os.path.join(output_dir,
+                                  'hSystematics_R{}_{}_{}-{}{}'.format(self.utils.remove_periods(jetR),
+                                                                       obs_label, int(min_pt_truth),
+                                                                       int(max_pt_truth), self.file_format))
     c.SaveAs(outputFilename)
     c.Close()
 
     sys_root_filename = os.path.join(output_dir, 'fSystematics.root')
     fSystematics = ROOT.TFile(sys_root_filename, 'UPDATE')
     h_total.Write()
     fSystematics.Close()
-      
+
   #----------------------------------------------------------------------
   # Add a list of (identically-binned) histograms in quadrature, bin-by-bin
   #----------------------------------------------------------------------
   def add_in_quadrature(self, h_list):
-  
+
     h_new = h_list[0].Clone()
     h_new.SetName('{}_new'.format(h_list[0].GetName()))
-    
+
     for i in range(1, h_new.GetNbinsX()+1):
-    
+
       values_i = [h.GetBinContent(i) for h in h_list]
 
       new_value_squared = 0.
       for value_i in values_i:
         new_value_squared += value_i*value_i
       new_value = math.sqrt(new_value_squared)
       h_new.SetBinContent(i, new_value)
-    
+
     return h_new
-    
+
+  #----------------------------------------------------------------------
+  def determine_reg_param_final(self, jetR, obs_label, obs_setting, grooming_setting,
+                                min_pt_truth, max_pt_truth):
+
+    # Obtain the total systematic uncertainty histograms from memory
+    min_reg_param = 3
+    reg_params = range(min_reg_param, self.max_reg_param + 1)
+    names = [ "hSystematic_Total_R{}_{}_n{}_{}-{}".format(self.utils.remove_periods(jetR), obs_label,
+                                                          reg_param, int(min_pt_truth), int(max_pt_truth))
+              for reg_param in reg_params ]
+    hSys_list = [ getattr(self, name) for name in names ]
+
+    # Obtain the statistical uncertainty histograms from disk
+    fdir = self.output_dir_main
```

Comment:
A useful trick is to do `os.path.join(fdir, 'myfile.root')`

Reply:
Thanks! Didn't know that took care of `'/'`

Reply:
Resolved in `c9433d0`